lib/std/os.zig

```diff
@@ -5215,6 +5215,60 @@ pub fn getFdPath(fd: fd_t, out_buffer: *[MAX_PATH_BYTES]u8) RealPathError![]u8 {
             };
             return target;
         },
+        .freebsd => {
+            var mib = [4]c_int{ CTL.KERN, KERN.PROC, KERN.PROC_FILEDESC, freebsd.getpid() };
+            var out_len: usize = 0;
+            try sysctl(&mib, null, &out_len, null, 0);
+
+            var kinfo_file: *freebsd.kinfo_file = undefined;
+
+            // allocate memory that is 64 bit alligned
+            var kinfo_file_buf = try std.heap.c_allocator.alloc(u64, (out_len / 8) + 1);
+            defer std.heap.c_allocator.free(kinfo_file_buf);
+
+            kinfo_file = @ptrCast(*freebsd.kinfo_file, kinfo_file_buf);
+
+            while (true) {
+                sysctl(&mib, @ptrCast(*anyopaque, kinfo_file), &out_len, null, 0) catch |err| switch (err) {
+                    error.SystemResources => {
+                        // increase memory size and try again
+                        out_len = (out_len * 4) / 3;
+                        kinfo_file_buf = try std.heap.c_allocator.realloc(kinfo_file_buf, (out_len / 8) + 1);
+                        kinfo_file = @ptrCast(*freebsd.kinfo_file, kinfo_file_buf);
+                        continue;
+                    },
+
+                    else => return err,
```

Comment:
Should be `else => |e| return e,` to remove `SystemResources` from the error set.